velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -253,10 +254,12 @@ struct DecimalAddSubtractBase {
   }
 
   // Computes the result precision and scale for decimal add and subtract
-  // operations following Hive's formulas.
+  // operations following Hive's formulas when `allowPrecisionLoss` is true.
   // If result is representable with long decimal, the result
   // scale is the maximum of 'aScale' and 'bScale'. If not, reduces result scale
   // and caps the result precision at 38.
+  // Caps p and s at 38 when not allowing precision loss.
```

Comment:
Do we `Caps p and s at 38` when allowPrecisionLoss being true or false?

Reply:
Not allow precision and scale loss mode just `Caps p and s at 38` while another one may reduce the scale.